readthedocs/proxito/views/hosting.py

```diff
@@ -0,0 +1,103 @@
+"""Views for hosting features."""
+
+import structlog
+from django.conf import settings
+from django.http import JsonResponse
+from django.views import View
+
+from readthedocs.builds.constants import EXTERNAL
+from readthedocs.core.mixins import CDNCacheControlMixin
+from readthedocs.core.unresolver import unresolver
+
+log = structlog.get_logger(__name__)  # noqa
+
+
+class ReadTheDocsConfigJson(CDNCacheControlMixin, View):
+    def get(self, request):
+
+        unresolved_domain = request.unresolved_domain
+        project = unresolved_domain.project
+
+        # TODO: why the UnresolvedURL object is not injected in the `request` by the middleware.
+        # Is is fine to calculate it here?
+        unresolved_url = unresolver.unresolve_url(request.headers.get("Referer"))
+        version = unresolved_url.version
+
+        # TODO: use Referrer header or GET arguments for Version / Build
+        project.get_default_version()
+        build = version.builds.last()
+
+        # TODO: define how it will be the exact JSON object returned here
+        # NOTE: we could use the APIv3 serializers for some of these objects if we want to keep consistency.
+        # However, those may require some extra db calls that we probably want to avoid.
+        data = {
```

Comment:
It seems like we're re-creating the footer API, but with JSON instead of HTML. That seems reasonable, but I do think we'll want to think more about what this API is returning.

It seems like we actually want the client to be able to tell us exactly what data points they want? Something similar to what we're doing with apiv3 with the `expand` and `fields` [parameters](https://docs.readthedocs.io/en/stable/api/v3.html#resources)? That way we give the caller control over how much data they want...

That might be an over-optimization at this stage, but I think we want to design this with expandability and performance in mind.

Reply:
> I do think we'll want to think more about what this API is returning.

Yes. This is one part of the design decisions I mentioned we have to sit down, talk together, think about and decide what's the best structure.

> It seems like we actually want the client to be able to tell us exactly what data points they want? Something similar to what we're doing with apiv3 with the expand and fields [parameters](https://docs.readthedocs.io/en/stable/api/v3.html#resources)? That way we give the caller control over how much data they want...

This is a good point to start considering and thinking about, yeah. However, I don't think we will require that complexity because:

* the API endpoint will be cached
* most of the values are static
* there are a few db queries only (`Project` and `Project.versions(active=True)`)
* we could decide whether or not return some specific data based on the feature being enabled. If `non_latest_version_warning` is _disabled_, we don't calculate/query nor return the list of active versions this feature consumes in the client.

In any case, I think we will arrive at the best answer by exploration, experimentation and brainstorming. Also, once we deploy this, we will have some data in NR to know how much time this endpoint takes on a real production db.

Reply:
To be clear, I'm fine shipping this as a test for now. 

But I do think over time if this is the "one endpoint to get all the data you need in the page", there will be demands on this functionality that are resource intensive, vary by user, etc. That is where we'll start to get value from the API approach.

Reply:
Good point about the .json name in the URL endpoint. We should definitely change that as first step.

I will think a little more about the expand argument. I don't have a strong opinion on that yet 😅